deepmd/utils/tabulate.py

```diff
@@ -82,20 +82,31 @@ def __init__(self,
         self.sub_graph, self.sub_graph_def = self._load_sub_graph()
         self.sub_sess = tf.Session(graph = self.sub_graph)
 
-        try:
-            self.sel_a = self.graph.get_operation_by_name('ProdEnvMatA').get_attr('sel_a')
-            self.prod_env_mat_op = self.graph.get_operation_by_name ('ProdEnvMatA')
-        except Exception:
-            self.sel_a = self.graph.get_operation_by_name('DescrptSeA').get_attr('sel_a')
-            self.prod_env_mat_op = self.graph.get_operation_by_name ('DescrptSeA')
+        if isinstance(self.descrpt, deepmd.descriptor.DescrptSeR):
+            try:
+                self.sel_a = self.graph.get_operation_by_name('ProdEnvMatR').get_attr('sel')
+                self.prod_env_mat_op = self.graph.get_operation_by_name ('ProdEnvMatR')
+            except Exception:
+                self.sel_a = self.graph.get_operation_by_name('DescrptSeR').get_attr('sel')
+                self.prod_env_mat_op = self.graph.get_operation_by_name ('DescrptSeR')
+        else:
```

Comment:
`elif isinstance(self.descrpt, deepmd.descriptor.DescrptSeA):` is recommended. 

And the conditions stop when `else` which capture the case self.descrpt is neither DescrptSeA nor DescrptSeR, and throw an error.

Reply:
Yes, it‘s safer. I will change it~